src/main/java/org/folio/circulation/resources/PickSlipsReportResource.java

```diff
@@ -0,0 +1,153 @@
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.CollectionResourceClient;
+import org.folio.circulation.support.CqlQuery;
+import org.folio.circulation.support.ItemRepository;
+import org.folio.circulation.support.OkJsonResponseResult;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.RouteRegistration;
+import org.folio.circulation.support.http.client.Response;
+import org.folio.circulation.support.http.server.WebContext;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import static org.folio.circulation.domain.ItemStatus.PAGED;
+import static org.folio.circulation.support.AsyncCoordinationUtil.allOf;
+import static org.folio.circulation.support.CqlQuery.exactMatch;
+import static org.folio.circulation.support.CqlQuery.exactMatchAny;
+import static org.folio.circulation.support.utils.BatchProcessingUtil.partitionList;
+
+public class PickSlipsReportResource extends Resource {
+
+  /**
+   * The optimal number of identifiers that will not exceed the permissible length
+   * of the URI according to the RFC 2616
+   */
+  private static final int UUID_BATCH_SIZE = 40;
+
+  private static final String SERVICE_POINT_ID_PARAM = "servicePointId";
+  private static final String STATUS_NAME_KEY = "status.name";
+  private static final String REQUEST_STATUS_KEY = "status";
+  private static final String REQUEST_ITEM_ID_KEY = "itemId";
+  private static final String REQUESTS_KEY = "requests";
+  private static final String ITEMS_KEY = "items";
+  private static final String TOTAL_RECORDS_KEY = "totalRecords";
+
+  private final String rootPath;
+
+  public PickSlipsReportResource(String rootPath, HttpClient client) {
+    super(client);
+    this.rootPath = rootPath;
+  }
+
+  @Override
+  public void register(Router router) {
+    RouteRegistration routeRegistration = new RouteRegistration(rootPath, router);
+    routeRegistration.getMany(this::getMany);
+  }
+
+  private void getMany(RoutingContext routingContext) {
+    final UUID servicePointId = UUID.fromString(routingContext.request().getParam(SERVICE_POINT_ID_PARAM));
+
+    final WebContext context = new WebContext(routingContext);
+    final Clients clients = Clients.create(context, client);
+
+    final CollectionResourceClient requestsStorageClient = clients.requestsStorage();
+    final ReportRepository reportRepository = new ReportRepository(clients);
+    final ItemRepository itemRepository = new ItemRepository(clients, true, false, false);
+
+    reportRepository.getAllItemsByField(STATUS_NAME_KEY, PAGED.getValue())
+        .thenApply(r -> r.next(this::mapFetcherToItems))
+        .thenComposeAsync(r -> r.after(items -> filterRequestedItems(items, requestsStorageClient)))
+        .thenComposeAsync(r -> r.after(items -> allOf(items, itemRepository::fetchItemRelatedRecords)))
```

Comment:
As far as I see `fetchItemRelatedRecords` method fetches:

* HoldingsRecord
* Instance
* Location
* MaterialType
* LoanType

But In a fact, you need `location` only. Doses it make sense fetch only location, for example by using `org.folio.circulation.domain.LocationRepository#getLocation`. 

An alternative can be to fetch all locations that have required service point id, and than build an extended item query (proper status and `effectiveLocationId` is any of the locations). But I'm aware that this story is a priority for Q4, so it's fine to skip this comment.